test/TestApps/PushNotificationsDemoApp/main.cpp

```diff
@@ -0,0 +1,146 @@
+            }
+        });
+
+    auto result = co_await channelOperation;
+
+    if (result.Status() == PushNotificationChannelStatus::CompletedSuccess)
+    {
+        auto channelUri = result.Channel().Uri();
+
+        std::cout << "channelUri: " << winrt::to_string(channelUri.ToString()) << std::endl << std::endl;
+
+        auto channelExpiry = result.Channel().ExpirationTime();
+
+        // Register Push Event for Foreground
+        result.Channel().PushReceived([](const auto&, PushNotificationReceivedEventArgs const& args)
+            {
+                auto payload = args.Payload();
+
+                // Do stuff to process the raw payload
+                std::string payloadString(payload.begin(), payload.end());
+                std::cout << "Push notification content received from FOREGROUND: " << payloadString << std::endl << std::endl;
+                args.Handled(true);
+            });
+        // Caller's responsibility to keep the channel alive
+        co_return result.Channel();
+    }
+    else if (result.Status() == PushNotificationChannelStatus::CompletedFailure)
+    {
+        LOG_HR_MSG(result.ExtendedError(), "We hit a critical non-retryable error with channel request!");
+        co_return nullptr;
+    }
+    else
+    {
+        LOG_HR_MSG(result.ExtendedError(), "Some other failure occurred.");
+        co_return nullptr;
+    }
+
+};
+
+winrt::Microsoft::Windows::PushNotifications::PushNotificationChannel RequestChannel()
+{
+    auto task = RequestChannelAsync();
+    if (task.wait_for(std::chrono::seconds(300)) != AsyncStatus::Completed)
+    {
+        task.Cancel();
+        return nullptr;
+    }
+
+    auto result = task.GetResults();
+    return result;
+}
+
+int main()
+{
+    time_t ltime;
+    char buf[SIZE];
+
+    time(&ltime);
+    ctime_s(buf, sizeof buf, &ltime);
+    std::cout << "Project Reunion Push Notification Test App: " << buf << std::endl;
+
+    PushNotificationActivationInfo info(
+        PushNotificationRegistrationOptions::PushTrigger | PushNotificationRegistrationOptions::ComActivator,
+        winrt::guid("ccd2ae3f-764f-4ae3-be45-9804761b28b2")); // same clsid as app manifest
+
+    auto token = PushNotificationManager::RegisterActivator(info);
+
+    auto args = AppInstance::GetCurrent().GetActivatedEventArgs();
+    auto kind = args.Kind();
+    if (kind == ExtendedActivationKind::Push)
+    {
+        PushNotificationReceivedEventArgs pushArgs = args.Data().as<PushNotificationReceivedEventArgs>();
+
+        // Call GetDeferral to ensure that code runs in low power
+        auto deferral = pushArgs.GetDeferral();
+
+        auto payload = pushArgs.Payload();
+
+        // Do stuff to process the raw payload
+        std::string payloadString(payload.begin(), payload.end());
+        std::cout << "Push notification content received from BACKGROUND: " << payloadString << std::endl << std::endl;
+        std::cout << "Press 'Enter' to exit the App." << std::endl;
```

Comment:
You know `printf` isn't that bad a thing :-)

(alas, no `std::format()` until C++20 => https://madridccppug.github.io/posts/stdformat/)

CONSIDER: Using printf instead of std::cout